### PR TITLE
ACCESS-ESM1.6 SBD

### LINKED ISSUE #140: Add ACCESS-ESM1.6 SBD
I would like to add an initial SBD for ACCESS-ESM1.6. For now, the only difference from the ACCESS-ESM1.5 SBD will be the MOM5 spec.

In ACCESS-ESM1.6, I would like to build MOM5 in the same way as ACCESS-ESM1.5, except **using the "master" branch and the "ACCESS-ESM" "type" variant**. To allow this, the following changes are required:

In MOM5 SPD:
- Add an "access-esm1.6" version that uses the "master" branch
- Remove the [`conditional` logic](https://github.com/ACCESS-NRI/spack-packages/blob/3f45685e8f4b2e10d03db5bcd0027bc22df475a4/packages/mom5/package.py#L32) from the definition of the "type" variant
- Extend `with`s and `if`s currently set up for `@access-esm1.5` to include `@access-esm1.6`

In ACCESS-ESM1.5 SPD:
- Specify `+type=ACCESS-CM` in MOM5 spec

In (new) ACCESS-ESM1.6 SPD:
- Specify `+type=ACCESS-ESM` in MOM5 spec

I'll follow up with a PR shortly.

### PR DESCRIPTION
**Pull request by dougiesquire**
This PR adds a new SBD for ACCESS-ESM1.6. At the moment, this is only differs from ACCESS-ESM1.5 in how MOM5 is built.

Closes #140

### CODE REVIEW

File: packages/mom5/package.py
Comment (penguian) [inline packages/mom5/package.py:490]: Should this be
```
if not self.spec.satisfies("@access-esm1.5:access-esm1.6"):
```
because of the `not` on line 493 of the previous?

**Review by penguian — CHANGES_REQUESTED** [review]
See my comment on line 490.

File: packages/mom5/package.py
Comment (dougiesquire) [reply packages/mom5/package.py:490]: Whoops, good catch. I was ~not~ paying enough attention when I changed the syntax. I think it should be

```
if not self.spec.satisfies("@access-esm1.5")
```

because the `master` and `access-esm1.6` versions both use the master branch and should enter here.

I'm away from my computer at the mo, but will change this when I'm back.

**Comment by dougiesquire** [comment]
I think I need approval from @penguian to allow the merge

**Comment by CodeGat** [comment]
Admins can dismiss requested changes, if you think he would be happy - just since he's not in this morning

**Comment by dougiesquire** [comment]
I've resolved his review comment, so I think he'd be happy?

**Comment by CodeGat** [comment]
Done @dougiesquire

**Pull request merged**
